tests/cpp/test_gpu_tensorcore.cpp

```diff
@@ -3222,6 +3222,289 @@ TEST_F(GPUTTensorCoreTest, MisalignedVectorization) {
   }
 }
 
+// Matmul test with multiple M dimensions that are consecutive
+TEST_F(GPUTTensorCoreTest, MultipleConsecutiveMDims) {
+  int M1 = 126, N = 136, M2 = 4, K = 248;
+
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  // Note that M2 is inside K, so this is an NN layout
+  auto tv0 = makeContigTensor(3, DataType::Half); // M1, M2, K
+  auto tv1 = makeContigTensor(2, DataType::Half); // N, K
+
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+
+  // M1, M2, N, K
+  tv0 = broadcast(tv0, {false, false, true, false});
+  tv1 = broadcast(tv1, {true, true, false, false});
+
+  auto tv2 = fusedMultiplySum(tv0, tv1, {-1});
+
+  fusion.addOutput(tv2);
+
+  MatMulTileOptions gemm_tile;
+  gemm_tile.cta_tile = GemmTile(128, 128, 32);
+  gemm_tile.warp_tile = GemmTile(64, 64, 32);
+  gemm_tile.instruction_tile = GemmTile(16, 8, 16);
+
+  MatmulParams params;
+  // Supported vec size is based on inner dim
+  params.supported_vec_size = {4, 8, 4};
+  params.mma_macro = MmaMacro::Ampere_16_8_16;
+  params.tile_sizes = gemm_tile;
+  params.async_gmem_load_operands = true;
+  params.double_buffer_options.double_buffer_smem_write = true;
+  params.double_buffer_options.double_buffer_smem_read = true;
+  params.double_buffer_options.smem_double_buffer_stage = 4;
+  scheduleMatmul(&fusion, params);
+
+  auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
+  at::Tensor A = at::randn({M1, M2, K}, options);
+  at::Tensor B = at::randn({N, K}, options);
+  std::vector<c10::IValue> inputs{A, B};
+
+  FusionExecutor fe;
+  NVFUSER_TEST_CUDA_ARCH_COMPILE_CHECK(
+      8, 0, fe.compileFusion(&fusion, inputs, LaunchParams(), matmul_cparams));
+  ASSERT_TRUE(getBankConflictInfo(fe.kernel()).empty());
+  auto cg_outputs = fe.runFusion(inputs);
+  auto tref = at::linear(A.to(at::kFloat), B.to(at::kFloat));
+  NVF_CHECK(cg_outputs[0].allclose(tref, 0.0001, 0.0001));
+}
+
+// Matmul test with multiple M dimensions that are non-consecutive
+// TODO: This test currently fails, but it can be run using
+//   build/test_matmul --gtest_also_run_disabled_tests
+//
+// This case fails because the allocation domain of the B smem cached tensor is
+// [N1, K, N2], which is incompatible with ldMatrix. We need to gather the
+// allocation domains for the smem tensors by role instead, so that this
+// becomes [K, N1, N2].
+TEST_F(GPUTTensorCoreTest, DISABLED_MultipleNonConsecutiveMDims) {
+  int M1 = 126, N = 136, M2 = 4, K = 248;
+
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  // Note that M2 is inside K, so this is an NN layout
+  auto tv0 = makeContigTensor(3, DataType::Half); // M1, K, M2
+  auto tv1 = makeContigTensor(2, DataType::Half); // N, K
+
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+
+  // M1, N, K, M2
```

Comment:
Should this be `M1, N, M2, K`?

Reply:
Actually, this is M1, N, K, M2, it's just that I screwed up the definition. I think I will combine this with the other disabled test and have non-consecutive, M, N, and K dims. Similarly I'll combine the enabled consecutive tests